Helper/Data.php

```diff
@@ -95,33 +131,84 @@ public function shouldCheckCaptcha()
     }
 
     /**
-     * Check  Smaily Cron Sync is enabled
+     * Check Smaily Cron Sync is enabled.
      *
      * @return bool
      */
-    public function isCronEnabled()
+    public function isCronEnabledForWebsite($websiteId)
     {
-        return (bool) $this->getGeneralConfig('enableCronSync');
+        return (bool) $this->getSmailyConfig('enableCronSync', $websiteId);
     }
 
     /**
-     * Check  Smaily Abandoned Cart is enabled
+     * Check Smaily Abandoned Cart is enabled.
      *
      * @return bool
      */
-    public function isAbandonedCartEnabled()
+    private function isAbandonedCartEnabledForWebsite($websiteId)
     {
-        return (bool) $this->getGeneralConfig('enableAbandonedCart');
+        return (bool) $this->getSmailyConfig('enableAbandonedCart', $websiteId);
     }
 
     /**
-     * Get Magento main configuration by field
+     * Get Magento configuration by field and website Id.
      *
      * @return string
      */
-    public function getConfigValue($configPath, $storeId = null)
+    public function getConfigValue($configPath, $websiteId = null)
+    {
+        return $this->scopeConfig->getValue($configPath, ScopeInterface::SCOPE_WEBSITE, $websiteId);
+    }
+
+    /**
+     * Get Website ID by current State area.
+     *
+     * @return int Website ID
+     */
+    private function getCurrentWebsiteId()
+    {
+        // Admin area
+        if ($this->state->getAreaCode() === \Magento\Framework\App\Area::AREA_ADMINHTML) {
+            $websiteId = (int) $this->request->getParam('website', 0);
+            return (int) $websiteId;
+        }
+        // Public area
+        $websiteId = $this->storeManager->getStore()->getWebsiteId();
+        return (int) $websiteId;
+    }
+
+    /**
+     * Get Website ID by Store ID.
+     *
+     * @param int|string Store ID
+     * @return int Website ID
+     */
+    private function getWebsiteIdByStoreID($storeId)
     {
-        return $this->scopeConfig->getValue($configPath, ScopeInterface::SCOPE_STORE, $storeId);
+        return (int) $this->storeManager->getStore($storeId)->getWebsiteId();
+    }
+
+    /**
+     * Get all Website IDs
+     *
+     * @return array Website IDs
+     */
+    public function getWebsiteIds()
+    {
+        $ids = [];
+        foreach ($this->websiteCollectionFactory->create() as $website) {
+            $ids[] = (int) $website->getId();
+        }
+        return $ids;
+    }
+
+    private function getWebsiteIdForWebsiteName($websiteName) {
+        foreach ($this->websiteCollectionFactory->create() as $website) {
+            if ($website->getName() === $websiteName) {
+                return (int) $website->getId();
+            }
+        }
+        return null;
```

Comment:
Rather than trying to fetch Website IDs from their names, add a `website_id` field to subscribers dataset, just don't synchronize the `website_id` field to Smaily.

Otherwise newsletter subscribers synchronization will become extremely slow and resource intensive.

Reply:
Added in commit. Unsetting `website_id` before sending to Smaily